components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -135,6 +143,11 @@ class GeckoEngineSession(
             session: GeckoSession?,
             securityInfo: GeckoSession.ProgressDelegate.SecurityInformation?
         ) {
+            // Ignore initial load of about:blank (see https://github.com/mozilla-mobile/android-components/issues/403)
+            if (initialLoad && securityInfo?.origin?.startsWith(MOZ_NULL_PRINCIPAL) == true) {
+                return
+            }
```

Comment:
If in your tests `onSecurityChange` happened after `onLocationChange`... will this block ever get executed?

Reply:
@pocmo yes, it will be executed for about:blank. Here's the flow of events for a single open tab:

OnLocationChange (about:blank) -> ignore (initialLoad = true)
OnLocationChange (about:blank) -> ignore (initialLoad = true)
OnSecurityChange (about:blank, origin:moz-nullprincipal) -> ignore (initialLoad = true)
OnLocationChange (taburl) -> notify (initialLoad = false)
OnSecurityChange (taburl, origin:taburl) -> notify (initialLoad = false)

Reply:
Thanks. Got it!